packages/base/limit.js

```diff
@@ -12,6 +12,8 @@ export default function limit(limit) {
       count++
       if (count <= limit) {
         yield chunk
+      } else {
+        break
```

Comment:
I don't know if it's feasible, but we could try to inject some conditional logic at startup, so that the `break` statement is executed only if there is a next step in the pipeline.
Anyway, I would attempt this in a separate PR, I think we can merge this for now

Reply:
That's an intriguing idea. Right now a step does not know whether it's last or not but sounds like something we can do

Alternatively, maybe a dummy `PassThrough` added ad the end of the pipeline would also solve this without any side effects?

Reply:
still, these are all workarounds